iOS/Source/MainController.swift

```diff
@@ -1,121 +1,236 @@
     }
-}
 
-extension MainController: LocationTrackerDelegate {
+    func screenshot() -> UIImage {
+        let margin: CGFloat = 20
+        let height: CGFloat = 450
+        let daylightLabelWidth: CGFloat = 100
 
-    func locationTracker(_ locationTracker: LocationTracker, didFailWith error: Error) {
-        guard Location.current == nil else { return }
+        let screenShotInsets = UIEdgeInsets(top: 32, left: 32, bottom: 32, right: 32)
 
-        let isUnexpectedError = (error as NSError).code != 0
-        if isUnexpectedError {
-            let alertController = UIAlertController(title: "Error", message: error.localizedDescription, preferredStyle: .alert)
-            alertController.addAction(UIAlertAction(title: "Dismiss", style: .default, handler: nil))
-            self.present(alertController, animated: true, completion: nil)
-        }
-    }
+        let y = self.messageLabel.frame.maxY - height + margin
+
+        let overlayView = UIView(frame: CGRect(x: 0, y: y, width: self.view.frame.width, height: height))
+        self.view.addSubview(overlayView)
+
+        self.informationButtonTopAnchor?.isActive = false
+        let screenshotInformationTopAnchor = self.informationButton.top(to: overlayView, offset: -18)
+
+        self.locationLabelBottomAnchor?.isActive = false
+        self.locationLabelLeftAnchor?.isActive = false
+        self.locationLabelRightAnchor?.isActive = false
+
+        let screenshotLocationCenterYAnchor = self.locationLabel.topAnchor.constraint(equalTo: self.informationButton.centerYAnchor)
+        screenshotLocationCenterYAnchor.isActive = true
+
+        let screenshotLocationRightAnchor = self.locationLabel.right(to: self.view, offset: -screenShotInsets.right)
+        let screenshotLocationLeftAnchor = self.locationLabel.left(to: self.view, offset: daylightLabelWidth + screenShotInsets.right)
+
+        self.locationLabel.textAlignment = .right
 
-    func locationTracker(_ locationTracker: LocationTracker, didFindLocation placemark: CLPlacemark) {
-        Location.current = Location(placemark: placemark)
+        self.informationButton.nameLabel.text = NSLocalizedString("Daylight", comment: "")
+        self.informationButton.nameLabel.sizeToFit()
 
-        self.rootView.updateInterface(location: Location.current)
+        let screenshot = UIScreen.screenshot(frame: overlayView.frame)
+
+        self.informationButton.nameLabel.text = NSLocalizedString("About", comment: "")
+
+        self.locationLabel.textAlignment = .left
+        screenshotLocationLeftAnchor.isActive = false
+        screenshotLocationRightAnchor.isActive = false
+        screenshotLocationCenterYAnchor.isActive = false
+
+        self.locationLabelBottomAnchor?.isActive = true
+        self.locationLabelLeftAnchor?.isActive = true
+        self.locationLabelRightAnchor?.isActive = true
+
+        screenshotInformationTopAnchor.isActive = false
+        self.informationButtonTopAnchor?.isActive = true
```

Comment:
Is there a way to shove all the overlay stuff into one view so you don't have to do all this crazy nonsense with activating and deactiviating constraints? Can wait for another PR

Reply:
Yeah that is good plan! Really want to get it out of here. But not sure how too, since it is based on views that are on the MainController :/

Reply:
Is there stuff that can be screenshotted and layered or something?